pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -96,3 +103,115 @@ func TestTaskRef(t *testing.T) {
 		})
 	}
 }
+
+func TestGetTaskFunc(t *testing.T) {
+	// Set up a fake registry to push an image to.
+	s := httptest.NewServer(registry.New())
+	defer s.Close()
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testcases := []struct {
+		name        string
+		localTasks  []runtime.Object
+		remoteTasks []runtime.Object
+		ref         *v1beta1.TaskRef
+		expected    runtime.Object
+	}{
+		{
+			name: "remote-task",
+			localTasks: []runtime.Object{
+				tb.Task("simple", tb.TaskType(), tb.TaskNamespace("default"), tb.TaskSpec(tb.Step("something"))),
+			},
+			remoteTasks: []runtime.Object{
+				tb.Task("simple", tb.TaskType()),
+				tb.Task("dummy", tb.TaskType()),
+			},
+			ref: &v1alpha1.TaskRef{
+				Name:   "simple",
+				Bundle: u.Host + "/remote-task",
+			},
+			expected: tb.Task("simple", tb.TaskType()),
+		}, {
+			name: "local-task",
+			localTasks: []runtime.Object{
+				tb.Task("simple", tb.TaskType(), tb.TaskNamespace("default"), tb.TaskSpec(tb.Step("something"))),
+			},
+			remoteTasks: []runtime.Object{
+				tb.Task("simple", tb.TaskType()),
+				tb.Task("dummy", tb.TaskType()),
+			},
+			ref: &v1alpha1.TaskRef{
+				Name: "simple",
+			},
+			expected: tb.Task("simple", tb.TaskType(), tb.TaskNamespace("default"), tb.TaskSpec(tb.Step("something"))),
+		}, {
+			name: "remote-cluster-task",
+			localTasks: []runtime.Object{
+				tb.ClusterTask("simple", tb.ClusterTaskType(), tb.ClusterTaskSpec(tb.Step("something"))),
+			},
+			remoteTasks: []runtime.Object{
+				tb.ClusterTask("simple", tb.ClusterTaskType()),
+				tb.ClusterTask("dummy", tb.ClusterTaskType()),
+			},
+			ref: &v1alpha1.TaskRef{
+				Name:   "simple",
+				Kind:   v1alpha1.ClusterTaskKind,
+				Bundle: u.Host + "/remote-cluster-task",
+			},
+			expected: tb.ClusterTask("simple", tb.ClusterTaskType()),
+		}, {
+			name: "local-cluster-task",
+			localTasks: []runtime.Object{
+				tb.ClusterTask("simple", tb.ClusterTaskType(), tb.ClusterTaskSpec(tb.Step("something"))),
+			},
+			remoteTasks: []runtime.Object{
+				tb.ClusterTask("simple", tb.ClusterTaskType()),
+				tb.ClusterTask("dummy", tb.ClusterTaskType()),
+			},
+			ref: &v1alpha1.TaskRef{
+				Name: "simple",
+				Kind: v1alpha1.ClusterTaskKind,
+			},
+			expected: tb.ClusterTask("simple", tb.ClusterTaskType(), tb.ClusterTaskSpec(tb.Step("something"))),
+		},
+	}
```

Comment:
These tests are great and helps understand whats going on in `taskref`. Can we also add few more tests with `v1beta1` Tasks/ClusterTasks?

Reply:
I believe the second and fourth test cover v1beta1 Tasks/ClusterTasks. Is there a specific case you wanted to see?